### ISSUE TITLE
Middle Click + Drag Multicursor

### ISSUE THREAD

**Issue by donatj** [zed-industries/zed#11990]
### Check for existing issues

- [X] Completed

### Describe the feature

I would really like it if middle click + drag was added for creating multiple cursors and multiline selections similar to how Sublime, VSCode and Intellij editors do so.

I was frankly kind of surprised I couldn't find a feature request for it.

### If applicable, add mockups / screenshots to help present your vision of the feature

Here's a little video of it working in Sublime Text

https://github.com/zed-industries/zed/assets/133747/eb53034c-02e3-4034-984b-742be9d71708

**Comment by mikayla-maki** [comment]
Interesting, so it acts like a drag but each new line becomes it's own multi cursor...

**Comment by donatj** [comment]
Yep. It's kind of similar to selecting a box of text by holding down alt and dragging in Terminal.app except each selected line gets its own cursor on whatever side the drag ended

|                                                                                                                              |                                                                                                                              | 
|------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------| 
| <img width="242" alt="image" src="https://github.com/zed-industries/zed/assets/133747/942d5886-9e18-4e1d-8e88-37c5e27447bc"> | <img width="204" alt="image" src="https://github.com/zed-industries/zed/assets/133747/e4bf170e-4e81-4798-ae83-201d4eb8251a"> |

**Comment by mikayla-maki** [comment]
Ahh! This feature already exists, under `alt-shift-left-mouse-drag`. I'll see about adding a middle click binding for it :)

**Comment by donatj** [comment]
Dang that was FAST! Thank you so much!

**Comment by cisoun** [comment]
Hi! First, thanks for the feature!

I've just noticed the selection has to be made in weird way though. To make it work, this is what I have to do:
1) Middle click and release
2) Left click and drag

My expectation was: middle click -> drag

Is this intended?

Zed preview 0.137.0, MacOS

**Comment by major-hmhco** [comment]
I am also observing a similar issue - simply middle-clicking and dragging does not activate multicursor select; I have to hold middle-click _while left-click/dragging_ to get a multicursor selection.

Zed preview 0.137.0, MacOS 12.7.4, 2021 MBP w/ M1 Pro

**Comment by CurbaiCode** [comment]
I am also experiencing a strange behavior. If I open a new file and `alt-shift-left-mouse-drag`, it acts as if the starting position for the command is the beginning of the file (the default new cursor location) and _not_ at the location the mouse was at when the command was called.

```
Zed: v0.137.5 (Zed)
OS: macOS 12.6.6
Memory: 16 GiB
Architecture: x86_64
```

**Comment by donatj** [comment]
@mikayla-maki I'd been patiently waiting for this to appear in a release, and it never seemed to materialize. 

Curious just now, I circled back to this ticket to ask when I could see it in a release and saw the comments. As is, it seems you have to hold down the middle click **and** left click to use the functionality?

That's unexpected and not quite what I wanted.

I was hoping for the behaviour most other editors have where you can just hold down middle click.

**Comment by mikayla-maki** [comment]
Oh? That sounds like a bug @donatj, thanks for letting me know.

**Comment by cisoun** [comment]
May I also add some feedback?

I guess the behaviour copies what VSCode does but I find the usage of Sublime Text easier here:

VSCode:
 - Needs SHIFT+OPT to activate.
 - Multicursor selection starts at the last cursor position, even if you clicked down somewhere else after SHIFT+OPT.

Sublime Text:
 - Need only OPT to activate.
 - Multicursor selection starts where you click down after OPT.

A mode to switch between the two would be nice, what do you think?

**Comment by davidcelis** [comment]
Adding another request for this feature; while having the `shift-opt-click -> drag` feature of VSCode is nice, it would also be really nice to have the `opt-click -> drag` feature, and they both behave differently.

* `shift-opt-click -> drag`: inserts multiple cursors and selects on each line, starting from where the cursor originally was, immediately moving to wherever the `shift-opt-click` occurred
* `opt-click -> drag`: inserts multiple cursors _starting from where you opt-click, regardless of where the cursor may have originally been_, adding cursors and selecting as you drag.

We have the first, but I heavily rely on (and would love to see) the latter!

**Comment by nikolamilekic** [comment]
On the new Windows version this works as expected for me (simple middle-click and drag), but on macOS I have to do the unnatural left and middle-click and drag to get it to work. Would love to see this fixed!

**Comment by donatj** [comment]
@nikolamilekic Good detail, I've only ever tried it on mac and it's just never worked on Mac for me

**Comment by debuxxed** [comment]
> On the new Windows version this works as expected for me (simple middle-click and drag), but on macOS I have to do the unnatural left and middle-click and drag to get it to work. Would love to see this fixed!

^ still present on mac

**Comment by marcharding** [comment]
Seems to work now on macOS 🙃
